### PR TITLE
Split out external address

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?
This splits out the external address from the wallet address.

This exposes the `balances`, `balance`, and `faucet` functionality on the external address.

```
address = Coinbase::ExternalAddress.new(:base_sepolia, '0x12345')
address.balances
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -286,6 +286,16 @@ w4 = u.wallet(w1.id)
 w4.load_seed(file_path)
 ```
 
+## External Addresses
+
+Addresses that you do not control the key for can still be interacted with via the Platform SDK.
```
Comment (jazz-cb) [inline README.md:291]: Addresses that do not belong CDP wallets can still be interacted with via the Platform SDK.
"Not control the key" isnt accurate. They could be controlling the key, just not loaded on platform sdk

File: lib/coinbase/address/external_address.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Coinbase
+  # A representation of a blockchain Address, which is a user-controlled account on a Network. Addresses are used to
+  # send and receive Assets. An ExternalAddress is an Address that is not controlled by the developer, but is instead
```
Comment (jazz-cb) [inline lib/coinbase/address/external_address.rb:5]: > An ExternalAddress is an Address that is not controlled by the developer, but is instead

This doesnt need to be true in all scenarios. ExternalAddress is an address that does not belong to a CDP Wallet

File: lib/coinbase/address/wallet_address.rb [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+      payloads = { signed_payload: trade.transaction.sign(@key) }
+
+      payloads[:approve_tx_signed_payload] = trade.approve_transaction.sign(@key) unless trade.approve_transaction.nil?
+
+      broadcast_trade(trade, **payloads)
+    end
+
+    # Returns whether the Address has a private key backing it to sign transactions.
+    # @return [Boolean] Whether the Address has a private key backing it to sign transactions.
+    def can_sign?
```
Comment (jazz-cb) [inline lib/coinbase/address/wallet_address.rb:108]: shouldnt this also be in the address class?

Comment (alex-stone) [reply lib/coinbase/address/wallet_address.rb:108]: Added!

**Review by jazz-cb — APPROVED** [review]

**Pull request merged**